README.md

```diff
@@ -94,31 +94,56 @@ Once you have implemented the tools, fill out the report below.
+
+We also came to the conclusion that making weekly exercises was a good idea since a large part of the project was already done.
+
 
 ### Problems encountered if any. 
+During implementation we found that the default buffersize of bufio.NewScanner is 2^16 bytes which caused problems during testing since when we wanted to test runningtime on genomes larger than 65536 characters the parser returned an empty string.
+
+In the beginning we also faced some problems with the folder structure in regards to using our implementations of naive and lin in testing and thus we resorted to creating a stub of both in the algos_test.go file to then use when testing.
 
-*Describe this here.*
 
 ### Experiments that verifies the correctness of your implementations.
 
-*Describe this here.*
+We verified the implementation of KMP by comparing the result to the result of the naive algorithm. We ran both algorithms on randomly generated data over different alphabets:  DNA, English alphabet, A, AB and (AA...AAB) and alphabets where occourences of some letters were way more common than others, etc. We also tried some edgecases such as empty x or p strings. In all cases the two algorithms generated reported the same matches. So we concluded that since both algorithms yielded some matches and the reads and genomes were sufficiently long and random, our implementation of KMP works.
+
 
 ### Experiments validating the running time.
 
 For this section, you should address the following:
 
 * An experiment that verifies that your implementation of `naive` uses no more time than O(nm) to find all occurrences of a given pattern in a text. Remember to explain your choice of test data. What are “best” and “worst” case inputs? 
 
+To show the `naive` algorithm runs in O(nm) we conducted tests with a variable x value and different fixed p values.
+These results can be seen in:
+
+![](figs/naive_p_constant.jpg)
+
+In this experiment we run the naive algorithm on a repetitive alphabet which is prone to containing a lot of matches before a mismatch occurs. This results in an expected worst case O(nm) performance.
+It can clearly be seen that an increase in p value from 100 to 200 results in linear growth in the running time. This can be seen since the running time roughly doubles between the two runs of the algorithm.
+
+We also tried to increment size of genome x and read p at the same time by some constant, in order to see if the algorithms running time would increase by n^2. This turned out to also be the case (notice that x axis is squared):
```

Comment:
Why would a function $f(n,m) \in O(nm)$ grow like $O(n^2)$ in this scenario? I am not sure I follow what you are doing.